plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -769,6 +787,25 @@ private static LongWriteFunction timestampSecondsWriteFunction(ClickHouseVersion
         };
     }
 
+    private static LongReadFunction shortTimestampWithTimeZoneReadFunction()
+    {
+        return (resultSet, columnIndex) -> {
+            OffsetDateTime offsetDateTime = resultSet.getObject(columnIndex, OffsetDateTime.class);
+            ZonedDateTime zonedDateTime = offsetDateTime.toZonedDateTime();
+            return packDateTimeWithZone(zonedDateTime.toInstant().toEpochMilli(), zonedDateTime.getZone().getId());
+        };
+    }
+
+    private static LongWriteFunction shortTimestampWithTimeZoneWriteFunction()
+    {
+        return (statement, index, value) -> {
+            long millisUtc = unpackMillisUtc(value);
+            TimeZoneKey timeZoneKey = unpackZoneKey(value);
+            OffsetDateTime offsetDateTime = OffsetDateTime.ofInstant(Instant.ofEpochMilli(millisUtc), timeZoneKey.getZoneId());
+            statement.setObject(index, offsetDateTime);
+        };
+    }
+
```

Comment:
both of these look like very standard impls that most databases should work with. Should we consider adding these to `StandardColumnMappings`?

cc: @findepi

Reply:
I think `resultSet.getObject(columnIndex, OffsetDateTime.class)` vs `resultSet.getObject(columnIndex, ZonedDateTime.class)` is the key. 
Generally the latter is preferred (named zones) and hopefylly works in most of the cases, if not all.
Would `resultSet.getObject(columnIndex, ZonedDateTime.class)` work for ClickHouse as well?

Reply:
@tangjiangling Can you check if `getObject(columnIndex, ZonedDateTime.class)` works for ClickHouse driver and adjust accordingly?

Reply:
I'll take a look.

Reply:
> Can you check if getObject(columnIndex, ZonedDateTime.class) works for ClickHouse driver

Yes, it works.